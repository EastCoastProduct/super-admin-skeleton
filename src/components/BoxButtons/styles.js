import { StyleSheet } from 'aphrodite/no-important';

export default StyleSheet.create({
  button: {
    display: 'inline-block',
    verticalAlign: 'middle',
    width: 150,
  },
  link: {
    display: 'inline-block',
    marginLeft: 35,
    verticalAlign: 'middle',
  },
});
