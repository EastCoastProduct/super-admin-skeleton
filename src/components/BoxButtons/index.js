import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import { css } from 'aphrodite/no-important';
import Button from '../Button';
import styles from './styles';

const BoxButtons = ({ children, disabled, link }) =>
  <div>
    <Button
      className={styles.button}
      type="submit"
      disabled={disabled}
    >{children}</Button>
    <Link className={css(styles.link)} to={link}>Cancel</Link>
  </div>;

BoxButtons.propTypes = {
  children: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  link: PropTypes.string.isRequired,
};

export default BoxButtons;
