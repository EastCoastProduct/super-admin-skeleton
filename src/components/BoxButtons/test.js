import React from 'react';
import { shallow } from 'enzyme';
import BoxButtons from './';

describe('BoxButtons component snapshot', () => {
  it('renders basic required data', () => {
    const wrapper = shallow(
      <BoxButtons disabled={false} link="/users">Create</BoxButtons>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders disabled and different link', () => {
    const wrapper = shallow(
      <BoxButtons disabled link="/user/1">Save</BoxButtons>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
