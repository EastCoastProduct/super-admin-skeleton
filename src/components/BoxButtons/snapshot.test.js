import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import BoxButtons from './';

describe('BoxButtons component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(
      <BoxButtons disabled={false} link="/users">Create</BoxButtons>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders when button is disabled', () => {
    const tree = renderer.create(
      <BoxButtons disabled={true} link="/user/1">Save</BoxButtons>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
