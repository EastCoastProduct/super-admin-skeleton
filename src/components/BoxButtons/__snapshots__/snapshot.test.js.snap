exports[`BoxButtons component snapshot renders basic required data 1`] = `
<div>
  <button
    className="button_w1w7v0-o_O-button_1q6mnjs"
    disabled={false}
    type="submit">
    Create
  </button>
  <a
    className="link_1mjr7jy"
    onClick={[Function]}
    style={Object {}}>
    Cancel
  </a>
</div>
`;

exports[`BoxButtons component snapshot renders when button is disabled 1`] = `
<div>
  <button
    className="button_w1w7v0-o_O-button_1q6mnjs"
    disabled={true}
    type="submit">
    Save
  </button>
  <a
    className="link_1mjr7jy"
    onClick={[Function]}
    style={Object {}}>
    Cancel
  </a>
</div>
`;
